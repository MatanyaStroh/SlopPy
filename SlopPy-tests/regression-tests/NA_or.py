''' 
  (NA or True) = True
  (True or NA) = True

  Sources: http://databases.about.com/cs/sql/a/aa042803a.htm
           http://www.sqlite.org/nulls.html
''' 

x = 1 / 0
assert type(x) is NA
assert (x or True) is True
assert (True or x) is True

assert (x or True or x) is True
assert (x or x or True) is True
assert (True or x or x) is True

