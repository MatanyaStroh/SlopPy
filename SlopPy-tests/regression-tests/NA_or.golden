{NA <type 'exceptions.ZeroDivisionError'> 'integer division or modulo by zero'
 Traceback (most recent call last):
  File "regression-tests/NA_or.py", line 9, in <module>
    x = 1 / 0

Locals: {'__builtins__': <module '__builtin__' (built-in)>, '__name__': '__main__', '__file__': 'regression-tests/NA_or.py', '__doc__': ' \n  (NA or True) = True\n  (True or NA) = True\n\n  Sources: http://databases.about.com/cs/sql/a/aa042803a.htm\n           http://www.sqlite.org/nulls.html\n', '__package__': None}
}
{JUMP_IF_TRUE(NA) no jump | regression-tests/NA_or.py <module>:11}
{JUMP_IF_TRUE(NA) no jump | regression-tests/NA_or.py <module>:14}
{JUMP_IF_TRUE(NA) no jump | regression-tests/NA_or.py <module>:15}
{JUMP_IF_TRUE(NA) no jump | regression-tests/NA_or.py <module>:15}
